src/sync/impl/sync_file.cpp

```
@@ -0,0 +1,305 @@
+        buffer.append(extension);
+    }
+    return buffer;
+}
+
+} // util
+
+constexpr const char SyncFileManager::c_sync_directory[];
+constexpr const char SyncFileManager::c_utility_directory[];
+constexpr const char SyncFileManager::c_metadata_directory[];
+constexpr const char SyncFileManager::c_metadata_realm[];
+
+std::string SyncFileManager::get_utility_directory()
+{
+    auto util_path = file_path_by_appending_component(get_base_sync_directory(),
+                                                      c_utility_directory,
+                                                      util::FilePathType::Directory);
+    util::try_make_dir(util_path);
+    return util_path;
+}
+
+std::string SyncFileManager::get_base_sync_directory()
+{
+    auto sync_path = file_path_by_appending_component(m_base_path,
+                                                      c_sync_directory,
+                                                      util::FilePathType::Directory);
+    util::try_make_dir(sync_path);
+    return sync_path;
+}
+
+std::string SyncFileManager::user_directory(const std::string& user_identity)
+{
+    REALM_ASSERT(user_identity.length() > 0);
+    auto user_path = file_path_by_appending_component(get_base_sync_directory(),
+                                                      user_identity,
+                                                      util::FilePathType::Directory);
+    util::try_make_dir(user_path);
+    return user_path;
+}
+
+void SyncFileManager::remove_user_directory(const std::string& user_identity)
+{
+    REALM_ASSERT(user_identity.length() > 0);
+    auto user_path = file_path_by_appending_component(get_base_sync_directory(),
+                                                      user_identity,
+                                                      util::FilePathType
+                                                      ::Directory);
+    try {
+        util::remove_nonempty_dir(user_path);
+    }
+    catch(File::AccessError) {
```

Why silently ignore `AccessError`?